operator/pkg/controller/istiocontrolplane/istiocontrolplane_controller.go

```diff
@@ -94,12 +97,24 @@ var (
 			return false
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			object, err := meta.Accessor(e.Object)
-			log.Debugf("got delete event for %s.%s", object.GetName(), object.GetNamespace())
+			obj, err := meta.Accessor(e.Object)
+			log.Debugf("got delete event for %s.%s", obj.GetName(), obj.GetNamespace())
 			if err != nil {
 				return false
 			}
-			if object.GetLabels()[helmreconciler.OwningResourceName] != "" && object.GetLabels()[helmreconciler.OwningResourceNamespace] != "" {
+			unsObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(e.Object)
+			if err != nil {
+				return false
+			}
+			if obj.GetLabels()[helmreconciler.OwningResourceName] != "" &&
```

Comment:
it would be better to extra this part into a helper function

Reply:
thanks, updated.